tests/unittests/benchmark/task/profet/test_profet_task.py

```diff
@@ -0,0 +1,375 @@
+            input_dir=profet_input_dir,
+            checkpoint_dir=checkpoint_dir,
+        )
+
+        results = second_task(point)
+
+        assert len(results) == 1
+        assert results[0]["type"] == "objective"
+        first_objective = results[0]["value"]
+        second_results = second_task(point)
+        second_objective = second_results[0]["value"]
+
+        assert second_objective == first_objective
+
+    def test_call_twice_with_same_task_gives_same_result(
+        self,
+        profet_train_config: MetaModelTrainingConfig,
+        profet_input_dir: Path,
+        checkpoint_dir: Path,
+    ):
+        """ When using the same task and different values for the same point, the results should
+        be identical.
+        """
+        task_id = 0
+        task = self.Task(
+            max_trials=10,
+            task_id=task_id,
+            train_config=profet_train_config,
+            seed=123,
+            input_dir=profet_input_dir,
+            checkpoint_dir=checkpoint_dir,
+        )
+
+        point = task._space.sample(1, seed=task.seed)[0]
+
+        first_results = task(point)
+        assert len(first_results) == 1
+        assert first_results[0]["type"] == "objective"
+        first_objective = first_results[0]["value"]
+
+        # Pass the same point under different forms and check that the results are exactly the same.
+        for second_point in [point, tuple(point), dict(zip(task._space.keys(), point))]:
+            second_results = task(second_point)
+            assert second_results[0]["name"] == first_results[0]["name"]
+            assert second_results[0]["type"] == first_results[0]["type"]
+            second_objective = second_results[0]["value"]
+
+            # TODO: Not sure why, but the two values are very close, but different!
+            assert np.isclose(first_objective, second_objective)
+
+    def test_call_with_gradients(
+        self,
+        profet_train_config: MetaModelTrainingConfig,
+        profet_input_dir: Path,
+        checkpoint_dir: Path,
+    ):
+        """ Test that calling the task with the `with_grad` returns the gradient at that point. """
+        task = self.Task(
+            max_trials=10,
+            task_id=0,
+            train_config=profet_train_config,
+            seed=123,
+            input_dir=profet_input_dir,
+            checkpoint_dir=checkpoint_dir,
+        )
+
+        point_tuple = task._space.sample(1)[0]
+        point_dict = dict(zip(task._space.keys(), point_tuple))
+        results = task(point_dict, with_grad=True)
+
+        assert results[0]["type"] == "objective"
+        first_objective = results[0]["value"]
+        assert results[1]["type"] == "gradient"
+        first_gradient = results[1]["value"]
+
+        # TODO: Check if this testing logic makes sense, and the last condition isn't yet working.
+        step_size = 1.0
+
+        second_point_array = np.array(point_tuple) - step_size * np.array(first_gradient)
```

Comment:
Perhaps you should compute the gradient using the metamodel directly instead?

Reply:
I don't think so, since I'm trying to create the target for what the gradient should be, it wouldn't make sense to use the underlying model to get the target as well, would it?
Maybe I'm missing something though, lmk.

Reply:
I don't think we need to test that pytorch is computing the gradient properly. What we want to test is that the task returns the correct gradient. So accessing what we know generate the correct gradients to double check would be fine. It's more like a simple sanity check.

Reply:
> test that pytorch is computing the gradient properly

This isn't what I'm doing here though! I'm just checking that the gradient I get makes sense.

> What we want to test is that the task returns the correct gradient.

This is what I'm (trying to) do here. For now though I just check the sign of the improvement. 
I'll leave this as-is.